### DISCUSSION TITLE
problem with redirect page - vite-plugin-ssr/client/router

### DISCUSSION THREAD

**Discussion by Nelie-Taylor** [vikejs/vike#796] [Discussion]
i have a problem with redirect page:
i user import { navigate } from 'vite-plugin-ssr/client/router'; for redirect (image).
But when it redirect to home page (/).
The system rerender page again.

`
useEffect(() => {
    }, []);
`
I don't know where am I going wrong.
Thank you for watch.
![image](https://user-images.githubusercontent.com/12018657/233431386-6cd5ab49-d799-4f5f-9a00-ee5545084c6a.png)

**Comment by Nelie-Taylor** [comment]
https://user-images.githubusercontent.com/12018657/233433066-49c44efd-b15d-4965-abff-2f9471ebd688.mp4

Watch my video

**Comment by brillout** [comment]
I'm not sure what the problem is? (I skimmed over the video; I usually prefer text over video.)

**Reply by Nelie-Taylor** [reply]
Hi Brillout, thank you for replay.
in short called, page/account.page.tsx

```
export const Page = () => {
  useEffect(() => {
   // function view load 1 time went access page
  }, []);
}
```

When i call ```navigate('/');```, website redirect to ```/```, useEffect in page/account.page.tsx be executed
i don't want useEffect in page/account.page.tsx to be executed when redirected by ```navigate('/')```;

**Reply by brillout** [reply]
Seems like `throw RenderErrorPage({ redirectTo: '/' })` could be a better approach, see https://vite-plugin-ssr.com/route-function#redirection-404.

**Reply by Nelie-Taylor** [reply]
![image](https://user-images.githubusercontent.com/12018657/233793831-0fee3109-ab81-4d9e-8e6e-59b76398c88a.png)
@brillout Looks like the interface is not valid.

**Reply by brillout** [reply]
It's a custom `pageContext.redirectTo` property, see the docs.

(Simple built-in redirection support is work-in-progress.)
